src/Raven.Server/Utils/ReplicationUtils.cs

```diff
@@ -13,6 +15,35 @@ namespace Raven.Server.Utils
 {
     internal static class ReplicationUtils
     {
+        private static readonly ConcurrentDictionary<string, (X509Certificate2, ClusterRequestExecutor)> _requestExecutorsCache = new ConcurrentDictionary<string, (X509Certificate2, ClusterRequestExecutor)>();
+
+        private static ClusterRequestExecutor GetOrCreateForSingleNode(string url, X509Certificate2 certificate)
+        {
+            ClusterRequestExecutor reqExecutor;
+            if (_requestExecutorsCache.ContainsKey(url))
+            {
+                reqExecutor = _requestExecutorsCache[url].Item2;
+                if (_requestExecutorsCache[url].Item1.Equals(certificate))
+                    return reqExecutor;
+
+                _requestExecutorsCache.Remove(url, out _);
+                reqExecutor.Dispose();
+            }
+
+            if (_requestExecutorsCache.Count > 128) // clear from time to time
+            {
+                while (_requestExecutorsCache.Count > 0)
+                {
+                    _requestExecutorsCache.Remove(url, out var certAndRequestor);                    
+                    certAndRequestor.Item2.Dispose();
+                }
+            }
+
+            reqExecutor = ClusterRequestExecutor.CreateForSingleNode(url, certificate);
+            _requestExecutorsCache[url] = (certificate, reqExecutor);
```

Comment:
Not concurrently safe

Reply:
Thread safety : https://github.com/ravendb/ravendb/pull/9871/commits/4c6e96b6306919736303fa7e2c9c0b2ef0438c07